src/S3/SSECMiddleware.php

```diff
@@ -63,8 +63,12 @@ private function prepareSseParams(CommandInterface $command, $prefix = '')
     {
         // Base64 encode the provided key
         $key = $command[$prefix . 'SSECustomerKey'];
-        $command[$prefix . 'SSECustomerKey'] = base64_encode($key);
-
+        if($command['SSECustomerKeyIsBase64']) {
```

Comment:
This check should verify that the field is `!empty()`, not just check if it's truthy, as its current state will give off . It also needs to work for all `$prefix` values.